jaxdecomp/_src/transpose.py

```diff
@@ -141,7 +154,10 @@ def infer_sharding_from_operands(kind: str, mesh: Mesh,
                                    arg_infos: Tuple[ShapeDtypeStruct],
                                    result_infos: Tuple[ShapedArray]):
     input_sharding = arg_infos[0].sharding
```

Comment:
Every step the processor grid is being switched from col major to row major
A test is added for this